File: iza-os-intelligence/lib/scoring.py
class ScoringEngine:
    def calculate_readiness(self, code_completeness, api_readiness, documentation_score, test_coverage):
        raw_score = (
            code_completeness * 0.4 +
            api_readiness * 0.3 +
            documentation_score * 0.2 +
            test_coverage * 0.1
        )
        score = round(raw_score)
        
        # Mapped from AGENTS.md rules
        if score >= 90:
            tier = 'AUTONOMOUS'
        elif score >= 80:
            tier = 'SUPERVISED'
        elif score >= 70:
            tier = 'MONITORED'
        else:
            tier = 'TRAINING'
            
        return {"score": score, "tier": tier}

    def generate_report(self, code_completeness, api_readiness, documentation_score, test_coverage):
        res = self.calculate_readiness(code_completeness, api_readiness, documentation_score, test_coverage)
        score = res["score"]
        tier = res["tier"]
        
        lines = [
            f"📊 Venture Readiness Audit: {score}% ({tier})",
            f"==========================================",
            f"- Code Completeness: {code_completeness}%",
            f"- API Gateway Wiring: {api_readiness}%",
            f"- Documentation Depth: {documentation_score}%",
            f"- Test Suite Coverage: {test_coverage}%",
            ""
        ]
        
        if tier == 'AUTONOMOUS':
            lines.append("✅ System is 100% autonomous. Ready for direct deployment without supervision.")
        elif tier == 'SUPERVISED':
            lines.append("⚠️ Requires team lead approval ($1K+) or manual oversight for transactions.")
        elif tier == 'MONITORED':
            lines.append("⚠️ Requires manager/director approval on all executions.")
        else:
            lines.append("❌ Undergoing training. Pull from production until score reaches 70%.")
            
        return "\n".join(lines)

File: iza-os-intelligence/lib/scoring.ts
/**
 * Venture Readiness and Codebase Evaluation Rubrics
 */

export interface VentureMetrics {
  codeCompleteness: number;  // 0 to 100
  apiReadiness: number;      // 0 to 100
  documentationScore: number;// 0 to 100
  testCoverage: number;      // 0 to 100
}

export class ScoringEngine {
  /**
   * Calculates overall venture readiness based on four core criteria
   */
  calculateReadiness(metrics: VentureMetrics): {
    score: number;
    tier: 'TRAINING' | 'MONITORED' | 'SUPERVISED' | 'AUTONOMOUS';
  } {
    const rawScore = (
      metrics.codeCompleteness * 0.4 +
      metrics.apiReadiness * 0.3 +
      metrics.documentationScore * 0.2 +
      metrics.testCoverage * 0.1
    );

    const score = Math.round(rawScore);
    
    // Auto-approve authority limits mapped from AGENTS.md rules
    let tier: 'TRAINING' | 'MONITORED' | 'SUPERVISED' | 'AUTONOMOUS';
    if (score >= 90) {
      tier = 'AUTONOMOUS';
    } else if (score >= 80) {
      tier = 'SUPERVISED';
    } else if (score >= 70) {
      tier = 'MONITORED';
    } else {
      tier = 'TRAINING';
    }

    return { score, tier };
  }

  /**
   * Generates a feedback report with improvement guidelines
   */
  generateReport(metrics: VentureMetrics): string {
    const { score, tier } = this.calculateReadiness(metrics);
    
    const lines = [
      `📊 Venture Readiness Audit: ${score}% (${tier})`,
      `==========================================`,
      `- Code Completeness: ${metrics.codeCompleteness}%`,
      `- API Gateway Wiring: ${metrics.apiReadiness}%`,
      `- Documentation Depth: ${metrics.documentationScore}%`,
      `- Test Suite Coverage: ${metrics.testCoverage}%`,
      ``,
    ];

    if (tier === 'AUTONOMOUS') {
      lines.push('✅ System is 100% autonomous. Ready for direct deployment without supervision.');
    } else if (tier === 'SUPERVISED') {
      lines.push('⚠️ Requires team lead approval ($1K+) or manual oversight for transactions.');
    } else if (tier === 'MONITORED') {
      lines.push('⚠️ Requires manager/director approval on all executions.');
    } else {
      lines.push('❌ Undergoing training. Pull from production until score reaches 70%.');
    }

    return lines.join('\n');
  }
}
